### PR TITLE
[ML-937] Refactoring of the method names in the standard graph library

### PR DESCRIPTION
**Pull request by thunterdb**
And some small fixes here and there.

### CODE REVIEW

**Comment by thunterdb** [comment]
cc @jkbradley

File: python/graphframes/graphframe.py [hunk 1]
```diff
@@ -170,10 +170,11 @@ def stronglyConnectedComponents(self, num_iterations):
         :param num_iterations: the number of iterations to run.
         :return:
         """
-        jgf = self._jvm_graph.stronglyConnectedComponents().setNumIterations(num_iterations).run()
+        jgf = self._jvm_graph.stronglyConnectedComponents().numIterations(num_iterations).run()
         return _from_java_gf(jgf, self._sqlContext)
 
-    def svdPlusPlus(self, conf_dict):
+    def svdPlusPlus(self, rank = 10, max_iterations = 2, min_value = 0.0, max_value = 5.0,
```
Comment (jkbradley) [inline python/graphframes/graphframe.py]: Use camelCase in Python too

File: src/main/scala/org/graphframes/lib/AggregateMessages.scala [hunk 2]
```diff
@@ -31,19 +31,19 @@ import org.graphframes.GraphFrame
...
  *    messages sent to each vertex.  It also runs the aggregation, computing a DataFrame
  *    with one row for each vertex which receives > 0 messages.  The DataFrame has 2 columns:
  *     - vertex column ID (named [[GraphFrame.ID]])
  *     - aggregate from messages sent to vertex (with the name given to the [[Column]] specified
- *       in [[AggregateMessages.Builder.agg()]])
+ *       in [[AggregateMessages.agg()]])
  *
  * When specifying the messages and aggregation function, the user may reference columns using:
- *  - [[AggregateMessages.src]]: column for source vertex of edge
+ *  - [[AggregateMessages]]: column for source vertex of edge
```
Comment (jkbradley) [inline src/main/scala/org/graphframes/lib/AggregateMessages.scala]: should not have changed

File: src/main/scala/org/graphframes/lib/AggregateMessages.scala [hunk 3]
```diff
@@ -55,81 +55,81 @@ import org.graphframes.GraphFrame
...
+      case (None, Some(toDst)) => toDst
+      case _ =>
+        // Should never happen. Specify this case to avoid compilation warnings.
+        throw new RuntimeException("AggregateMessages: No messages were specified to be sent.")
     }
+    unionedMsgs.groupBy(ID).agg(aggCol)
   }
+}
+
+private[graphframes] object AggregateMessages extends Logging with Serializable {
```
Comment (jkbradley) [inline src/main/scala/org/graphframes/lib/AggregateMessages.scala]: Needs to be public to make src, dst, etc. accessible.

File: src/main/scala/org/graphframes/lib/BFS.scala [hunk 4]
```diff
@@ -86,9 +86,35 @@ import GraphFrame.nestAsCol
...
+
+  def maxPathLength(value: Int): this.type = {
+    require(value >= 0, s"BFS maxPathLength must be >= 0, but was set to $value")
+    maxPathLength = value
+    this
+  }
+
+  private var edgeFilter: Option[Column] = None
+
+  def setEdgeFilter(value: Column): this.type = {
```
Comment (jkbradley) [inline src/main/scala/org/graphframes/lib/BFS.scala]: rename: edgeFilter

File: src/main/scala/org/graphframes/lib/BFS.scala [hunk 5]
```diff
@@ -86,9 +86,35 @@ import GraphFrame.nestAsCol
...
+    edgeFilter = Some(value)
+    this
+  }
+
+  def setEdgeFilter(value: String): this.type = setEdgeFilter(expr(value))
```
Comment (jkbradley) [inline src/main/scala/org/graphframes/lib/BFS.scala]: ditto

File: src/main/scala/org/graphframes/lib/PageRank.scala [hunk 6]
```diff
@@ -71,9 +71,48 @@ import org.graphframes.GraphFrame
...
+    resetProb = Some(p)
+    this
+  }
+
+  def untilConvergence(tolerance: Double): this.type = {
+    tol = Some(tolerance)
+    this
+  }
+
+  def fixedIterations(i: Int): this.type = {
```
Comment (jkbradley) [inline src/main/scala/org/graphframes/lib/PageRank.scala]: Let's call this ```numIter```  (to be eventually replaced by maxIter)

File: src/main/scala/org/graphframes/lib/PageRank.scala [hunk 7]
```diff
@@ -71,9 +71,48 @@ import org.graphframes.GraphFrame
...
+    numIters = Some(i)
+    this
+  }
+
+  def run(): GraphFrame = {
+    tol match {
+      case Some(t) =>
+        PageRank.runUntilConvergence(graph, t, resetProb.get, srcId)
```
Comment (jkbradley) [inline src/main/scala/org/graphframes/lib/PageRank.scala]: For this case, make sure that numIters is not set.

File: src/main/scala/org/graphframes/lib/SVDPlusPlus.scala [hunk 8]
```diff
@@ -24,99 +24,110 @@ import org.apache.spark.sql.types._
...
+  private var _gamma7: Double = 0.015
+
+  private var _loss: Option[Double] = None
+
+  def rank(rank: Int): this.type = {
+    _rank = rank
+    this
+  }
+
+  def maxIterations(iter: Int): this.type = {
```
Comment (jkbradley) [inline src/main/scala/org/graphframes/lib/SVDPlusPlus.scala]: rename: maxIter

File: src/main/scala/org/graphframes/lib/StronglyConnectedComponents.scala [hunk 9]
```diff
@@ -21,38 +21,39 @@ import org.apache.spark.graphx.{lib => graphxlib}
...
+ *  - id: the id of the vertex
+ *  - weight (double): the normalized weight (page rank) of this vertex.
+ *
+ * @param graph the graph for which to compute the SCC
+ * @return a graph with vertex attributes containing the smallest vertex id in each SCC
+ */
+class StronglyConnectedComponents private[graphframes] (private val graph: GraphFrame) extends Arguments {
+  private var numIters: Option[Int] = None
+
+  def numIterations(iterations: Int): this.type = {
```
Comment (jkbradley) [inline src/main/scala/org/graphframes/lib/StronglyConnectedComponents.scala]: ditto: numIter

**Comment by jkbradley** [comment]
Done with review.  Just minor comments.  Thanks!

File: python/graphframes/graphframe.py (hunk 1 above)
Comment (thunterdb) [inline python/graphframes/graphframe.py]: done

File: src/main/scala/org/graphframes/lib/AggregateMessages.scala (hunk 2 above)
Comment (thunterdb) [inline src/main/scala/org/graphframes/lib/AggregateMessages.scala]: For some reason, intellij is considering  them to be errors. Oh well.

File: src/main/scala/org/graphframes/lib/AggregateMessages.scala (hunk 3 above)
Comment (thunterdb) [inline src/main/scala/org/graphframes/lib/AggregateMessages.scala]: exposed

File: src/main/scala/org/graphframes/lib/BFS.scala (hunk 4 above)
Comment (thunterdb) [inline src/main/scala/org/graphframes/lib/BFS.scala]: I missed this one, thanks.

**Comment by thunterdb** [comment]
@jkbradley comments addressed (and all tests are passing).

File: src/main/scala/org/graphframes/lib/AggregateMessages.scala (hunk 2 above)
Comment (jkbradley) [inline src/main/scala/org/graphframes/lib/AggregateMessages.scala]: IntelliJ does not like links to a class' members within docs for the overall class.  I've never been sure how to get around that.

File: python/graphframes/graphframe.py [hunk 10]
```diff
@@ -121,33 +121,33 @@ def connectedComponents(self):
...
-        :param max_steps: the number of super steps to be performed.
+        :param maxSteps: the number of super steps to be performed.
         :return:
         """
-        jgf = self._jvm_graph.labelPropagation().setMaxSteps(max_steps).run()
+        jgf = self._jvm_graph.labelPropagation().maxSteps(maxSteps).run()
         return _from_java_gf(jgf, self._sqlContext)
 
-    def pageRank(self, reset_prob = 0.15, source_id = None, fixed_num_iter = None,
+    def pageRank(self, resetProb = 0.15, sourceId = None, fixedNumIter = None,
```
Comment (jkbradley) [inline python/graphframes/graphframe.py]: numIter  (I'll make a more general comment about this.)

File: python/graphframes/graphframe.py [hunk 11]
```diff
@@ -121,33 +121,33 @@ def connectedComponents(self):
...
         Note: Exactly one of fixed_num_iter or tolerance must be set.
-        :param reset_prob:
-        :param source_id: (optional) the source vertex for a personalized PageRank.
-        :param fixed_num_iter: If set, the algorithm is run for a fixed number
+        :param resetProb:
+        :param sourceId: (optional) the source vertex for a personalized PageRank.
+        :param fixedNumIter: If set, the algorithm is run for a fixed number
                of iterations. This may not be set if the `tolerance` parameter is set.
         :param tolerance: If set, the algorithm is run until the given tolerance.
                This may not be set if the `fixed_num_iter` parameter is set.
```
Comment (jkbradley) [inline python/graphframes/graphframe.py]: update param name

**Comment by jkbradley** [comment]
We should have the Python argument names match the Scala ones.  This also makes me think we should do one more update: The Scala setter method names should be the parameter names, and their arguments should be called "value."  That way, the Python API arguments can easily be matched to the Scala setters (so that the public API names match).

**Comment by jkbradley** [comment]
LGTM.  I'll merge this.  Thanks!

**Pull request merged**